evm-types.md

```diff
@@ -665,21 +665,31 @@ Accounts
 Storage/Memory Lookup
 ---------------------
 
-`#lookup*` looks up a key in a map and returns 0 if the key doesn't exist, otherwise returning its value.
+`#lookup*` looks up a key in the contract storage and returns 0 if the key doesn't exist, otherwise returning its value.
 
 ```k
-    syntax Int ::= #lookup        ( Map , Int ) [function, functional, smtlib(lookup)]
+    syntax ContractStorage ::= Map
+
+    syntax Int ::= #lookup        ( ContractStorage , Int ) [function, functional, smtlib(lookup)]
                  | #lookupMemory  ( Map , Int ) [function, functional, smtlib(lookupMemory)]
  // ----------------------------------------------------------------------------------------
-    rule [#lookup.some]:         #lookup(       (KEY |-> VAL:Int) _M, KEY ) => VAL modInt pow256
-    rule [#lookup.none]:         #lookup(                          M, KEY ) => 0                 requires notBool KEY in_keys(M)
+    rule [#lookup.some]:         #lookup(       (KEY |-> VAL:Int) _M:Map, KEY ) => VAL modInt pow256
+    rule [#lookup.none]:         #lookup(                          M:Map, KEY ) => 0                 requires notBool KEY in_keys(M)
     //Impossible case, for completeness
-    rule [#lookup.notInt]:       #lookup(       (KEY |-> VAL    ) _M, KEY ) => 0                 requires notBool isInt(VAL)
+    rule [#lookup.notInt]:       #lookup(       (KEY |-> VAL    ) _M:Map, KEY ) => 0                 requires notBool isInt(VAL)
 
-    rule [#lookupMemory.some]:   #lookupMemory( (KEY |-> VAL:Int) _M, KEY ) => VAL modInt 256
-    rule [#lookupMemory.none]:   #lookupMemory(                    M, KEY ) => 0                 requires notBool KEY in_keys(M)
+    rule [#lookupMemory.some]:   #lookupMemory( (KEY |-> VAL:Int) _M, KEY )     => VAL modInt 256
+    rule [#lookupMemory.none]:   #lookupMemory(                    M, KEY )     => 0                 requires notBool KEY in_keys(M)
     //Impossible case, for completeness
-    rule [#lookupMemory.notInt]: #lookupMemory( (KEY |-> VAL    ) _M, KEY ) => 0                 requires notBool isInt(VAL)
+    rule [#lookupMemory.notInt]: #lookupMemory( (KEY |-> VAL    ) _M, KEY )     => 0                 requires notBool isInt(VAL)
+```
+
+`#write` stores a value for a certain key in the contract storage.
+
+```k
+    syntax ContractStorage ::= #write ( ContractStorage , Int, Int ) [function, functional]
+ // ---------------------------------------------------------------------------------------
+    rule #write ( M:Map , KEY, VAL) => M [ KEY <- VAL ]
```

Comment:
Merge this code block with the one above, and make the documentation into bullet lists. One bullet for `#lookup`, one for `#lookupMemory`, and one for `#write`.

Reply:
Done